fix(nuttx): incorrect draw buffer size for I1 color and bpp=1

mateiuli opened the pull request:
Fix incorrect draw buffer size in `lv_nuttx_lcd.c:lcd_init()` when color format is `I1` and `bpp=1`.

I also added some code to skip the color palette in `flush_cb()` before calling `ioctl(..., LCDDEVIO_PUTAREA,...)`. I'm not sure if this is the correct approach, maybe NuttX's `drivers/lcd/lcd_dev.c` must handle the palette information. 

Is it like a standard for `I1-I8` color formats to prepend the palette before the draw buffer? If so, then I think NuttX's lcd driver must handle it. 

### Description of Testing
I tested this with NuttX + LVGL configured to use LCD_DEV on a custom board with 1.54" e-INK display with SSD1681 controller.
You can check out my topic on LVGL forum where I investigated and found this fix, [topic here](https://forum.lvgl.io/t/lvgl9-nuttx-eink-cannot-render-anything-useful/20283). 
 

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

liamHowatt reviewed (commented):
Nice work! To support I1 display sizes where width not divisible by 8 (not sure how common it is), it should be `... * bpp + 7) / 8` to round up. `lv_draw_buf_width_to_stride` can be used.

mateiuli commented:
Thank you all for review! 

@liamHowatt, I have rebased on top of master to fix the conflicts. It appears that the issue with the buffer size was fixed in the meanwhile by the following commit: aa567970c50722bed3bfe050606399cd8614ea66

I only kept the changes from `flush_cb()` and incorporated your feedback to use `lv_draw_buf_width_to_stride()` :).

liamHowatt reviewed (approved):
Ah nice! Thank you!

> I also added some code to skip the color palette in `flush_cb()` before calling `ioctl(..., LCDDEVIO_PUTAREA,...)`. I'm not sure if this is the correct approach, maybe NuttX's `drivers/lcd/lcd_dev.c` must handle the palette information.
> 
> Is it like a standard for `I1-I8` color formats to prepend the palette before the draw buffer? If so, then I think NuttX's lcd driver must handle it.

I took a quick look at `drivers/lcd/lcd_framebuffer.c` which handles `PUTAREA` and is a layer above `drivers/lcd/ssd1680.c`. I don't see anything that tries to interpret a palette if `bpp < 8` :+1: I could be wrong.
